fix : updated "add_bucket_name_suffix" var to true

HarikaPonna opened the pull request:
### Description
Set add_bucket_name_suffix to true by default to prevent bucket name conflicts.
When upgrading existing deployments, set add_bucket_name_suffix = false to preserve existing bucket names and avoid recreation.
<!--- Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it. --->

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
Set add_bucket_name_suffix to true by default to prevent bucket name conflicts.
When upgrading existing deployments, set add_bucket_name_suffix = false to preserve existing bucket names and avoid recreation.
<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

Ak-sky commented:
@HarikaPonna, you also need to make a change here-https://github.com/terraform-ibm-modules/terraform-ibm-cos/blob/3d6dafb52a0f7d19827c728df136f0b391460ca5/modules/fscloud/variables.tf#L65
Please check thoroughly and make the changes accordingly.

ocofaigh commented:
Just realised Andrej's PR didnt update fscloud submodule so re-opening this..

ocofaigh reviewed (changes requested):
Oh looks like we need to make an update to the test. The test is not taking into consideration the suffix I guess:

```
pr_test.go:253: 
        	Error Trace:	/__w/terraform-ibm-cos/terraform-ibm-cos/tests/pr_test.go:253
        	Error:      	Should be true
        	Test:       	TestRunFSCloudExample
        	Messages:   	CBRs are Not working as expected. Bucket cos-fscloud-6nr-bucket can be accessed at https://cos-fscloud-6nr-bucket.s3.us-south.cloud-object-storage.appdomain.cloud/ with response code: 404 and body: <?xml version="1.0" encoding="UTF-8" standalone="yes"?><Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist.</Message><Resource>/cos-fscloud-6nr-bucket/</Resource><RequestId>2c8b4eb8-595d-4555-9519-037b5efdc70d</RequestId><httpStatusCode>404</httpStatusCode></Error>
CBRs are working as expected, blocked at https://cos-fscloud-6nr-bucket.s3.private.us-south.cloud-object-storage.appdomain.cloud/ with error. Bucket cos-fscloud-6nr-bucket is not exposed. Bucket cannot be accessed. error: Get "https://cos-fscloud-6nr-bucket.s3.private.us-south.cloud-object-storage.appdomain.cloud/": dial tcp 10.1.129.45:443: i/o timeout
CBRs are working as expected, blocked at https://cos-fscloud-6nr-bucket.s3.direct.us-south.cloud-object-storage.appdomain.cloud/ with error. Bucket cos-fscloud-6nr-bucket is not exposed. Bucket cannot be accessed. error: Get "https://cos-fscloud-6nr-bucket.s3.direct.us-south.cloud-object-storage.appdomain.cloud/": dial tcp 161.26.0.34:443: i/o timeout
```

HarikaPonna commented:
/run pipeline

HarikaPonna commented:
> @HarikaPonna As discussed, you need to fix the unit test to account for the suffix in the bucket name now

@ocofaigh I’ve addressed the issue and updated the unit test to handle the bucket name suffix.
